### PR TITLE
[front] fix date format (timestamp) in horizontal bars widget (#7921)

### LINKED ISSUE #7921: [Dashboard] Date displayed as non-human readable format (timestamp)
## Description

When I make a widget breakdown by date, the dates are timestamps, which is unreadable/incomprehensible by a human. The dates should be displayed in another format (e.g. DD-MM-YYYY)

Screenshot 2024-08-01 155258

## Environment

OCTI 6.2.10

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create a widget horizontal bar
2. Entities perspective
3. Filter on "Entity type: Intrusion set"
4. Parameters like this: 

Screenshot 2024-08-01 155307

 5. Validated and looked at the dates

### PR DESCRIPTION
**Pull request by CelineSebe**
### Proposed changes

* change horizontalBarsChartOptions

### Related issues

* #7921 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Review by SarahBocognano — APPROVED** [review]
Tested ✅

File: opencti-platform/opencti-front/src/utils/Charts.js [hunk 1]
```diff
@@ -31,6 +32,7 @@ const toolbarOptions = {
       columnDelimiter: ',',
       headerCategory: 'category',
       headerValue: 'value',
+      // eslint-disable-next-line @typescript-eslint/no-shadow
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/utils/Charts.js:34]: Please do not use those disabling rules. Change the name of the argument instead

**Review by lndrtrbn — DISMISSED** [review]

File: opencti-platform/opencti-front/src/utils/Charts.js [hunk 2]
```diff
@@ -524,7 +526,17 @@ export const horizontalBarsChartOptions = (
     show: stackType !== '100%',
     labels: {
       show: stackType !== '100%',
-      formatter: (value) => (yFormatter && typeof value === 'number' ? yFormatter(value) : value),
+      formatter: (value) => {
+        if (yFormatter && typeof value === 'number') {
+          return yFormatter(value);
+        }
+        if (typeof value === 'string' && value.length === 15) {
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/utils/Charts.js]: What means this `value.length === 15` ?

File: opencti-platform/opencti-front/src/utils/Charts.js
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/utils/Charts.js:1]: In my opinion the fix should be done in the file calling the function `horizontalBarsChartOptions`, not in this file. We have the possibility already to pass a function `yFormatter` to format the value. So use this to transform the timestamp in date string when necessary directly in the widget component when the attribute is a date.

File: opencti-platform/opencti-front/src/utils/Charts.js (hunk 2 above)
Comment (CelineSebe) [reply opencti-platform/opencti-front/src/utils/Charts.js]: Timestamp has 15 characters...

**Pull request merged**